src/pacer.js

```diff
@@ -400,6 +412,17 @@ let PACER = {
     return !!(pacerCookie && !pacerCookie.match(/unvalidated/));
   },
 
+  // Given document.cookie, returns true if the user has filing rights.
+  hasFilingCooking: function (cookieString){
+    let cookies = {};
+    cookieString.replace(/\s*([^=;]+)=([^;]*)/g, function (match, name, value) {
+      cookies[name.trim()] = value.trim();
+    });
+    let filingCookie = cookies['filing']
+    console.log(filingCookie)
+    return !!(filingCookie && filingCookie.match(/true/));
+  },
```

Comment:
There's gotta be a (much) better way. JavaScript sometimes surprises me, but are you sure this is the best way to do this??

Reply:
You're right! I think We can use the approach shown in the [MDN Web Docs](https://developer.mozilla.org/en-US/). Here's the piece of code used to get the value of a cookie named test2:

```javascript
const cookieValue = document.cookie
  .split('; ')
  .find((row) => row.startsWith('test2='))
  ?.split('=')[1];
```

The whole example is [here](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie#example_2_get_a_sample_cookie_named_test2).

Reply:
Yeah, that's better, but man, JS just really sucks sometimes!